tests/qrcode.rs

```diff
@@ -1,43 +1,101 @@
 mod fixtures;
 
-use fixtures::{server, server_no_stderr, Error, TestServer};
+use assert_cmd::prelude::CommandCargoExt;
+use assert_fs::TempDir;
+use fixtures::{port, server_no_stderr, tmpdir, Error, TestServer};
 use reqwest::StatusCode;
 use rstest::rstest;
-use select::document::Document;
-use select::predicate::Attr;
-use std::iter::repeat_with;
+use std::process::Command;
+use std::thread::sleep;
+use std::time::Duration;
+
+#[cfg(not(windows))]
+fn run_in_faketty_kill_and_get_stdout(template: &Command) -> Result<String, Error> {
+    use std::process::Stdio;
+
+    use fake_tty::{bash_command, get_stdout};
+
+    let cmd = {
+        let bin = template.get_program().to_str().expect("not UTF8");
+        let args = template
+            .get_args()
+            .map(|s| s.to_str().expect("not UTF8"))
+            .collect::<Vec<_>>()
+            .join(" ");
+        format!("{} {}", bin, args)
+    };
+    let mut child = bash_command(&cmd).stdin(Stdio::null()).spawn()?;
+
+    sleep(Duration::from_secs(1));
+
+    child.kill()?;
+    let output = child.wait_with_output().expect("Failed to read stdout");
+    let all_text = get_stdout(output.stdout)?;
+
+    Ok(all_text)
+}
+
+#[rstest]
+// Disabled for Windows because `fake_tty` does not currently support it.
+#[cfg(not(windows))]
+fn qrcode_hidden_in_tty_when_disabled(tmpdir: TempDir, port: u16) -> Result<(), Error> {
+    let mut template = Command::cargo_bin("miniserve")?;
+    template.arg("-p").arg(port.to_string()).arg(tmpdir.path());
+
+    let output = run_in_faketty_kill_and_get_stdout(&template)?;
+
+    assert!(!output.contains("QR code for "));
+    Ok(())
+}
 
 #[rstest]
-fn hide_qrcode_element(server: TestServer) -> Result<(), Error> {
-    let body = reqwest::blocking::get(server.url())?.error_for_status()?;
-    let parsed = Document::from_read(body)?;
-    assert!(parsed.find(Attr("id", "qrcode")).next().is_none());
+// Disabled for Windows because `fake_tty` does not currently support it.
+#[cfg(not(windows))]
+fn qrcode_shown_in_tty_when_enabled(tmpdir: TempDir, port: u16) -> Result<(), Error> {
+    let mut template = Command::cargo_bin("miniserve")?;
+    template
+        .arg("-p")
+        .arg(port.to_string())
+        .arg("-q")
+        .arg(tmpdir.path());
+
+    let output = run_in_faketty_kill_and_get_stdout(&template)?;
 
+    assert!(output.contains("QR code for "));
     Ok(())
 }
 
 #[rstest]
-fn show_qrcode_element(#[with(&["-q"])] server: TestServer) -> Result<(), Error> {
-    let body = reqwest::blocking::get(server.url())?.error_for_status()?;
-    let parsed = Document::from_read(body)?;
-    assert!(parsed.find(Attr("id", "qrcode")).next().is_some());
+fn qrcode_hidden_in_non_tty_when_enabled(tmpdir: TempDir, port: u16) -> Result<(), Error> {
```

Comment:
This still prints a bunch of QR codes to my actual terminal while running tests. Is this intended? Can we capture the output here and process it without it showing up?

Reply:
Fixed. Test wasn't actually effective since I forgot to set `stdout` to `piped` mode.